### PR TITLE
Allow triggerRequest and signalCancel to be coalesced

### PR DESCRIPTION
**Pull request by jroper**
See #453 and https://github.com/reactive-streams/reactive-streams-jvm/pull/441#discussion_r288380996

### CODE REVIEW

File: tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java [hunk 1]
```diff
@@ -266,8 +266,9 @@ public void required_spec208_mustBePreparedToReceiveOnNextSignalsAfterHavingCall
       @Override
       public void run(WhiteboxTestStage stage) throws InterruptedException {
         stage.puppet().triggerRequest(1);
-        stage.puppet().signalCancel();
         stage.expectRequest();
+        stage.puppet().signalCancel();
+        stage.expectCancelling();
```
Comment (jroper) [inline tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java]: Now we expect the request before signalling cancel, preventing the coalescing of the request and cancel, and by expecting cancelling before signalling next, we ensure that signalling next does test what this test says it tests.

Comment (jrudolph) [reply tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java]: Sounds reasonable and less prone to race conditions.

**Comment by viktorklang** [comment]
@reactive-streams/contributors Feedback on this?

**Comment by akarnokd** [comment]
I'm not sure. Is this something because of Akka Stream's sequential signal count and mixing requests with cancellation in the same actor message queue?

File: tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java (hunk 1 above)
Comment (Scottmitch) [reply tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java]: +1. the behavior you described sounds like an improvement.

**Comment by viktorklang** [comment]
@akarnokd I think @jroper will be able to answer that. That aside though, the importance to the TCK is to be able to enforce the spec, so it is more about making sure of that.

**Comment by viktorklang** [comment]
@jroper @akarnokd @reactive-streams/contributors Let's make some progress here so we can release the first 1.0.3 RC. Feedback encouraged

**Comment by Scottmitch** [comment]
@viktorklang - can we merge this, cut an RC, and collect comments when folks kick the tires on the RC?

**Comment by viktorklang** [comment]
@Scottmitch Agreed—people have had ample time to respond as is. Will tackle this on Monday

**Pull request merged**